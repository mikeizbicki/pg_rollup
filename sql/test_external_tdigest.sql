create or replace language plpython3u;
drop extension pg_rollup;
create extension if not exists tdigest;
create extension if not exists pg_rollup;

create table tdigest_test (
    id serial primary key,
    a int
);

insert into tdigest_test (a) (select * from generate_series(0,10000));

select create_rollup(
    'tdigest_test',
    'tdigest_test_rollup1',
    rollups => $$
        tdigest(a)
    $$
);

select assert_rollup('tdigest_test_rollup1');

insert into tdigest_test (a) (select * from generate_series(0,10000));

SELECT assert_rollup_relative_error('tdigest_test_rollup1', 0.1);

insert into tdigest_test (a) (select * from generate_series(100000,200000));

SELECT assert_rollup_relative_error('tdigest_test_rollup1', 0.1);
