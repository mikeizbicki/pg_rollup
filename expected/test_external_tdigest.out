SET client_min_messages TO WARNING;
create or replace language plpython3u;
drop extension pgrollup;
create extension if not exists tdigest;
create extension if not exists pgrollup;
create table tdigest_test (
    id serial primary key,
    a int
);
insert into tdigest_test (a) (select * from generate_series(0,10000));
CREATE MATERIALIZED VIEW tdigest_test_rollup1 AS (
    SELECT
        tdigest_getpercentile(tdigest(a,10),0.50) AS p50,
        tdigest_getpercentile(tdigest(a,10),0.90) AS p90,
        tdigest_getpercentile(tdigest(a,10),0.99) AS p99,
        tdigest_getpercentile(tdigest(a,1000),0.50) AS p1000_50,
        tdigest_getpercentile(tdigest(a,1000),0.90) AS p1000_90,
        tdigest_getpercentile(tdigest(a,1000),0.99) AS p1000_99
    FROM tdigest_test
);
select assert_rollup('tdigest_test_rollup1');
 assert_rollup 
---------------
 
(1 row)

insert into tdigest_test (a) (select * from generate_series(0,10000));
SELECT assert_rollup_relative_error('tdigest_test_rollup1', 0.1);
 assert_rollup_relative_error 
------------------------------
 
(1 row)

insert into tdigest_test (a) (select * from generate_series(100000,200000));
SELECT assert_rollup_relative_error('tdigest_test_rollup1', 0.1);
 assert_rollup_relative_error 
------------------------------
 
(1 row)

drop table tdigest_test cascade;
